### PR TITLE
Remove SearchAbout command, deprecated in 8.5

### PR DESCRIPTION
**Pull request by jfehrle**
- [x] Entry added in the changelog.

### CODE REVIEW

**Comment by jfehrle** [comment]
I'm not 100% sure that HoTT_coq_010.v and bug_3900.v should be deleted.

File: doc/tools/docgram/fullGrammar
Comment (ppedrot) [inline doc/tools/docgram/fullGrammar]: This does not seem to belong here.

File: doc/tools/docgram/orderedGrammar
Comment (ppedrot) [inline doc/tools/docgram/orderedGrammar]: Same remark as above.

File: test-suite/bugs/closed/HoTT_coq_010.v
Comment (ppedrot) [inline test-suite/bugs/closed/HoTT_coq_010.v]: Use `Search` instead.

File: test-suite/bugs/closed/bug_3900.v
Comment (ppedrot) [inline test-suite/bugs/closed/bug_3900.v]: Use `Search` instead.

File: test-suite/success/searchabout.v
Comment (ppedrot) [inline test-suite/success/searchabout.v]: For the ones that make sense, use `Search` instead.

File: doc/tools/docgram/fullGrammar
Comment (Zimmi48) [reply doc/tools/docgram/fullGrammar]: This is just because `fullGrammar` and `orderedGrammar` are not automatically updated whenever the grammar is changed (for now). This is thus a follow-up of #11877. Note that these changes were automatically generated.

File: vernac/search.ml
Comment (Zimmi48) [inline vernac/search.ml]:
```suggestion
(** Search *)
```

File: vernac/ppvernac.ml
Comment (Zimmi48) [inline vernac/ppvernac.ml]: Not a big deal, but since you've renamed the `search_about` and `search_about_filter` functions, you could also rename the `pr_search_about` function.

File: doc/changelog/07-commands-and-options/11944-rm-searchabout-cmd.rst
Comment (Zimmi48) [inline doc/changelog/07-commands-and-options/11944-rm-searchabout-cmd.rst]:
```suggestion
- **Removed:** SearchAbout command that was deprecated in 8.5.
  Use :cmd:`Search` instead
  (`#11944 <https://github.com/coq/coq/pull/11944>`_, by Jim Fehrle).
```

**Comment by MSoegtropIMC** [comment]
This is not neccessarily an issue of Coq but of ssreflect, but when I have ssreflect loaded the normal `Search` behaves like `SearchHead` and I need to call `SearchAbout` when I want to have the effect `Search` should have. If this is removed I think I might be in trouble ... Can someone from ssreflect can comment, maybe @gares?

**Comment by anton-trunov** [comment]
@MSoegtropIMC You probably want to use `Search` with an underscore like so:
```
Search _ <your-usual-patttern>.
```
Please see this wiki page explaining some common gotchas of using the SSReflect version of `Search`:
https://github.com/math-comp/math-comp/wiki/Search

**Comment by MSoegtropIMC** [comment]
@anton-trunov : thanks, I guess this will work for the cases I felt I need SearchAbout.
@jfehrle : fine with me then.

**Comment by jfehrle** [comment]
Updated.

**Review by ppedrot — APPROVED** [review]

**Review by Zimmi48 — APPROVED** [review]

**Comment by Zimmi48** [comment]
I'll merge this PR tomorrow.

**Comment by llelf** [comment]
Sorry I‘m late to this party. But
- isn’t pg still uses `SearchAbout`?
- with ssr loaded, SearchAbout was a nice escape hatch to vanilla search.

**Comment by jfehrle** [comment]
> doesn’t pg still use `SearchAbout`?

`SearchAbout` has been deprecated since 8.5, which was released in 2016.  Seems like that's more than plenty of time to update PG.

> with ssr loaded, SearchAbout was a nice escape hatch to vanilla search.

Meaning a convenience that's still available through different syntax?
I'll leave it to others to respond to this point.

**Comment by Zimmi48** [comment]
This can always be reverted if the change is considered too problematic.
Regarding the SSReflect issue, have you noticed the answer to this question above?

**Comment by anton-trunov** [comment]
There is still some things that the SSReflect `Search` does not support, e.g. blacklisting.

**Comment by llelf** [comment]
> SearchAbout has been deprecated since 8.5, which was released in 2016. Seems like that's more than plenty of time to update PG.

@jfehrle  Well, guess what. They have not ;-)

> Regarding the SSReflect issue, have you noticed the answer to this question above?

@Zimmi48 Yes, but it does ssreflect search.  `SearchAbout` was a (the only??) way to escape to normal search.

**Comment by Zimmi48** [comment]
OK then, we've got two alternative options that I can see:
- restore `SearchAbout` as a synonym of `Search` but undeprecate it since we don't intend to remove it;
- finally add the missing features of SSReflect's `Search` to the default `Search` and remove SSReflect's `Search`. @herbelin has been interested in implementing these features and @gares has said he was OK to remove the non-tactic features from the SSReflect plugin (if we integrate them in the main code base). Cf. #8663 and the WIP PR #8855. Since `Search` is a query command changing the syntax from one version to the next shouldn't be an issue, but we can even provide a compatibility option to leave SSReflect users time to adapt to a new syntax. I'm afraid that the above mentioned WIP PR has been stuck because it was too ambitious. If its ambitions could scale down to just providing whatever SSReflect's `Search` provides, that would already be enough.

**Comment by herbelin** [comment]
Would adding an optional `concl:pattern` as first argument of vanilla `Search` be, e.g., an acceptable way to unify both `Search`?

**Comment by Zimmi48** [comment]
Personally, I don't know much about `SSReflect` so I'll leave to SSReflect users and maintainers to answer.

**Comment by cpitclaudel** [comment]
> > `SearchAbout` has been deprecated since 8.5, which was released in 2016. Seems like that's more than plenty of time to update PG.

>  @jfehrle Well, guess what. They have not ;-)

Fun fact: unless someone opens an issue on the PG side, or at least flags a maintainer, it's unlikely "they" will notice ^^.  Fixed in https://github.com/ProofGeneral/PG/commit/a38887058166487607f03972d6d2c25ee9d5dada.

**Comment by herbelin** [comment]
Actually, for basically the same price, we could add a modifier `hyps:pattern`. No objections?

**Comment by llelf** [comment]
> OK then, we've got two alternative options that I can see:

> restore SearchAbout as a synonym of Search but undeprecate it since we don't intend to remove it;

> finally add the missing features of SSReflect's Search to the default Search and remove SSReflect's Search. ⟨…⟩

@Zimmi48 or (until `Search` is unified):

3\) restore it, leave as deprecated, mention it in ssreflect section of the manual?

**Comment by herbelin** [comment]
I forgot that #8855 already provided a `concl` and an `hyp` option :(. I updated it so that it is closer to a mergeable PR.

**Comment by Zimmi48** [comment]
@llee454 In my quality of RM for 8.12, I will make sure that this issue is solved by the time of the release (preferably by ensuring #8855 gets merged in a state where it at least provide equivalent functionality to SSR's Search and deprecating SSR's Search and moving it to a separate module that needs to be explicitly imported).

**Pull request merged**
